In [19]:
import pandas as pd
df = pd.read_csv("C:/Users/DELL/Documents/FS2022_Data Science/Project 1.1 Analyzing Baby Names/names/yob2021.txt", header=None)

## Task 1:

In [22]:
# Inspect the file yob2021.txt in a text editor. What do you observe about the structure of the file?
df.head()
df.info

<bound method DataFrame.info of                0  1      2
0         Olivia  F  17728
1           Emma  F  15433
2      Charlotte  F  13285
3         Amelia  F  12952
4            Ava  F  12759
...          ... ..    ...
31532     Zyeire  M      5
31533       Zyel  M      5
31534      Zyian  M      5
31535      Zylar  M      5
31536        Zyn  M      5

[31537 rows x 3 columns]>

The file contains 3 columns with 31536 rows.
The columns are separated by Name, Sex, and a number of occurrences. 
There is no header on the top of each column, but the "readme" file describes what type of information each column contains.

## Task 2:

In [25]:
# Read the file 'yob2021.txt' into pandas.
df = pd.read_csv("C:/Users/DELL/Documents/FS2022_Data Science/Project 1.1 Analyzing Baby Names/names/yob2021.txt", header=None)

## Task 3:

In [26]:
# Print the first 10 rows.
print(df.head(10))

           0  1      2
0     Olivia  F  17728
1       Emma  F  15433
2  Charlotte  F  13285
3     Amelia  F  12952
4        Ava  F  12759
5     Sophia  F  12496
6   Isabella  F  11201
7        Mia  F  11096
8     Evelyn  F   9434
9     Harper  F   8388


## Task 4:

In [27]:
# Display the number of rows and columns.
df.shape

(31537, 3)

## Task 5:

In [30]:
# Calculate the total number of babies born in 2021, i.e. the sum of the third column.
total = df[2].sum()
print(total)

3361375


## Task 6:

In [32]:
# Like Task 5, but calculate the sum for boys and girls separately.
girls_sum = df[df[1] == "F"][2].sum()
print(girls_sum)

boys_sum = df[df[1] == "M"][2].sum()
print(boys_sum)

1627098
1734277


## Task 7:

In [33]:
# Check if your name occurs in the data.
df[df[0] == "Sara"]

0  1     2
195    Sara  F  1507
29346  Sara  M     6

## Task 8:

In [38]:
# Calculate the percentage of girls and boys among the total births.
girls_percent = girls_sum / total * 100
print(girls_percent)

boys_percent = boys_sum / total * 100
print(boys_percent)

48.405726823100665
51.59427317689933


## Task 9:

In [62]:
# Create a table that contains the top 5 girls names and top 5 boys names.
top_boys = df[df[1] == "M"].sort_values(2, ascending = False).head(5)
top_girls = df[df[1] == "F"].sort_values(2, ascending = False).head(5)
top = pd.concat([top_boys, top_girls])
print(top)

               0  1      2
17544       Liam  M  20272
17545       Noah  M  18739
17546     Oliver  M  14616
17547     Elijah  M  12708
17548      James  M  12367
0         Olivia  F  17728
1           Emma  F  15433
2      Charlotte  F  13285
3         Amelia  F  12952
4            Ava  F  12759


## Task 10:

In [72]:
# Write the data from task 9 to an Excel spreadsheet.
top.to_excel("C:/Users/DELL/Documents/FS2022_Data Science/Project 1.1 Analyzing Baby Names/top.xlsx")